cryptoconditions/types/ed25519.py

```diff
@@ -23,29 +36,69 @@ def __init__(self, public_key=None):
         which corresponds to a bitmask of 0x20.
 
         Args:
-            public_key (VerifyingKey): Ed25519 publicKey
-        """
-        if public_key and isinstance(public_key, (str, bytes)):
-            public_key = VerifyingKey(public_key)
-        if public_key and not isinstance(public_key, VerifyingKey):
-            raise TypeError
-        self.public_key = public_key
-        self.signature = None
-
-    def write_common_header(self, writer):
-        """
-        Write static header fields.
-
-        Some fields are common between the hash and the fulfillment payload. This
-        method writes those field to anything implementing the Writer interface.
-        It is used internally when generating the hash of the condition, when
-        generating the fulfillment payload and when calculating the maximum fulfillment size.
-
-        Args:
-            writer (Writer, Hasher, Predictor): Target for outputting the header.
-        """
-        writer.write_var_octet_string(self.public_key)
-
+            public_key (bytes): Ed25519 public key.
+            signature (bytes): Signature.
+            asn1 (): pyasn1 object.
+
+        """
+        if public_key is not None:
+            self._validate_public_key(public_key)
+        self._public_key = public_key
+        if signature is not None:
+            self._validate_signature(signature)
+        self._signature = signature
+
+    # TODO check type or use static typing (mypy)
+    def _validate_public_key(self, public_key):
+        if len(public_key) != self.PUBLIC_KEY_LENGTH:
+            raise Exception(
+                'Public key must be {} bytes, was: {}'.format(
+                    self.PUBLIC_KEY_LENGTH, len(public_key)))
+        # TODO More validation? Ask ILP folks.
```

Comment:
The underlying crypto library should already do extra checks. In fact this could just call `VerifyKey` and see if an exception is thrown.